File: LENTABP10/src/main/java/com/lenta/lentabp10/models/memory/MemoryTaskRepository.java
package com.lenta.lentabp10.models.memory;

import com.lenta.lentabp10.models.repositories.ITaskExciseStampRepository;
import com.lenta.lentabp10.models.repositories.ITaskProductRepository;
import com.lenta.lentabp10.models.repositories.ITaskRepository;
import com.lenta.lentabp10.models.repositories.ITaskWriteOffReasonRepository;

import lombok.Getter;

public class MemoryTaskRepository implements ITaskRepository {

    @Getter private ITaskProductRepository products;
    @Getter private ITaskExciseStampRepository exciseStamps;
    @Getter private ITaskWriteOffReasonRepository writeOffReasons;

    public MemoryTaskRepository(ITaskProductRepository taskProductRepository, ITaskExciseStampRepository taskExciseStampRepository, ITaskWriteOffReasonRepository taskWriteOfReasonRepository) {
        if (taskProductRepository == null)
        {
            throw new NullPointerException("taskProductRepository");
        }
        if (taskExciseStampRepository == null)
        {
            throw new NullPointerException("taskExciseStampRepository");
        }
        if (taskWriteOfReasonRepository == null)
        {
            throw new NullPointerException("taskWriteOfReasonRepository");
        }

        this.products = taskProductRepository;
        this.exciseStamps = taskExciseStampRepository;
        this.writeOffReasons = taskWriteOfReasonRepository;
    }

}

File: writeoff_breakage/src/main/java/com/lenta/bp10/models/memory/MemoryTaskRepository.kt
package com.lenta.bp10.models.memory

import com.lenta.bp10.models.repositories.ITaskExciseStampRepository
import com.lenta.bp10.models.repositories.ITaskProductRepository
import com.lenta.bp10.models.repositories.ITaskRepository
import com.lenta.bp10.models.repositories.ITaskWriteOffReasonRepository

class MemoryTaskRepository(
        private val taskProductRepository: ITaskProductRepository = MemoryTaskProductRepository(),
        private val taskExciseStampRepository: ITaskExciseStampRepository = MemoryTaskExciseStampRepository(),
        private val taskWriteOfReasonRepository: ITaskWriteOffReasonRepository = MemoryTaskWriteOffReasonRepository()
) : ITaskRepository {

    override fun getProducts(): ITaskProductRepository {
        return taskProductRepository
    }

    override fun getExciseStamps(): ITaskExciseStampRepository {
        return taskExciseStampRepository
    }

    override fun getWriteOffReasons(): ITaskWriteOffReasonRepository {
        return taskWriteOfReasonRepository
    }

}